components/dashboard/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC } from "react";
+import { ErrorBoundary, FallbackProps, ErrorBoundaryProps } from "react-error-boundary";
+import gitpodIcon from "../icons/gitpod.svg";
+
+export const GitpodErrorBoundary: FC = ({ children }) => {
+    return (
+        <ErrorBoundary FallbackComponent={DefaultErrorFallback} onReset={handleReset} onError={handleError}>
+            {children}
+        </ErrorBoundary>
+    );
+};
+
+export const DefaultErrorFallback: FC<FallbackProps> = ({ error, resetErrorBoundary }) => {
+    const emailSubject = encodeURIComponent("Gitpod Dashboard Error");
+    const emailBody = encodeURIComponent(`\n\nError: ${error.message}`);
```

Comment:
@selfcontained Can we also reference `error.code?` in the text and email body? :sparkles: 
Can totally happen as a follow-up.

@easyCZ A nice follow-up would be to extend `ResponseError` with `traceId`, and use that here as well. :bulb:
(Not saying we should to this now, was more of a realization. :lotus_position: )

Reply:
Digging into this, I don't believe we'll be bubbling up `ResponseError`s to an `ErrorBoundary`. These will be from runtime javascript errors thrown outside of event handlers, so no `code` property would be present.